### PR TITLE
TestScopedOperations does not work as expected

### PR DESCRIPTION
**Pull request by mcimadamore**
When Julia started to work on the `MemorySegment::overlap` API, she noticed (thanks!) an issue with the TestScopedOperation. Essentially this test checks that calling methods on various objects should be scope-safe - that is, it shouldn't be possible to call certain methods (e.g. MemorySegment::address) if the scope associated with the object is closed, or from a thread other than the owning thread.

The test had a fatal flaw, which was probably introduced when we moved to the `ResourceScope` API. Since the scoped operation creates the desired object (e.g. a segment) and then performs an operation on it, and since that operation is performed _after_ the scope has been closed, as a result, all tests trivially fail as expected, since the object cannot be created in the first place (the scope is closed!!).

The solution is to separate the logic for creating the object from the logic for testing the scoped operation. In fixing the test I realized that the test was running two combinations which didn't make sense (prefixAllocator::allocate, and segment::spliterator, neither of which is scoped), and found a bug in the VaList implementation - for VaList::skip, which was missing a scope check (which this patch also addresses).

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Reviewers
 * [Jorn Vernee](https://openjdk.java.net/census#jvernee) (@JornVernee - Committer) ⚠️ Review applies to 2e0e4ea0f3d41763734dde440660376cf2e3dc60

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/panama-foreign pull/587/head:pull/587` \
`$ git checkout pull/587`

Update a local copy of the PR: \
`$ git checkout pull/587` \
`$ git pull https://git.openjdk.java.net/panama-foreign pull/587/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 587`

View PR using the GUI difftool: \
`$ git pr show -t 587`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/panama-foreign/pull/587.diff">https://git.openjdk.java.net/panama-foreign/pull/587.diff</a>

</details>

### CODE REVIEW

**Comment by FrauBoes** [comment]
Looks good, and you found a bug along the way, nice.

File: src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/SegmentAllocator.java [hunk 1]
```diff
@@ -26,6 +26,7 @@
 package jdk.incubator.foreign;
 
 import jdk.internal.foreign.ArenaAllocator;
+import jdk.internal.foreign.ResourceScopeImpl;
```
Comment (JornVernee) [inline src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/SegmentAllocator.java:29]: Spurious import?

**Review by JornVernee — APPROVED** [review]
Looks like you have some merge conflicts as well

**Pull request closed without merging**